action/protocol/staking/protocol.go

```diff
@@ -650,40 +686,51 @@ func (p *Protocol) calculateVoteWeight(v *VoteBucket, selfStake bool) *big.Int {
 	return CalculateVoteWeight(p.config.VoteWeightCalConsts, v, selfStake)
 }
 
+type nonceUpdateType bool
+
+const (
+	updateNonce   nonceUpdateType = true
+	noUpdateNonce nonceUpdateType = false
+)
+
 // settleAction deposits gas fee and updates caller's nonce
 func (p *Protocol) settleAction(
 	ctx context.Context,
 	sm protocol.StateManager,
 	status uint64,
 	logs []*action.Log,
 	tLogs []*action.TransactionLog,
+	gasConsumed uint64,
+	updateNonce nonceUpdateType,
 ) (*action.Receipt, error) {
 	actionCtx := protocol.MustGetActionCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
-	gasFee := big.NewInt(0).Mul(actionCtx.GasPrice, big.NewInt(0).SetUint64(actionCtx.IntrinsicGas))
-	depositLog, err := p.depositGas(ctx, sm, gasFee)
+	gasFee := big.NewInt(0).Mul(actionCtx.GasPrice, big.NewInt(0).SetUint64(gasConsumed))
+	depositLog, err := p.helperCtx.DepositGas(ctx, sm, gasFee)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to deposit gas")
 	}
-	accountCreationOpts := []state.AccountCreationOption{}
-	if protocol.MustGetFeatureCtx(ctx).CreateLegacyNonceAccount {
-		accountCreationOpts = append(accountCreationOpts, state.LegacyNonceAccountTypeOption())
-	}
-	acc, err := accountutil.LoadAccount(sm, actionCtx.Caller, accountCreationOpts...)
-	if err != nil {
-		return nil, err
-	}
-	if err := acc.SetPendingNonce(actionCtx.Nonce + 1); err != nil {
-		return nil, errors.Wrap(err, "failed to set nonce")
-	}
-	if err := accountutil.StoreAccount(sm, actionCtx.Caller, acc); err != nil {
-		return nil, errors.Wrap(err, "failed to update nonce")
+	if updateNonce {
```

Comment:
so if the`createNFTBucket` executed successfully, the nonce is increased by EVM, then we don't increase nonce again? would that be considered the same nonce (actionCtx.Nonce + 1) did 2 actions: withdraw bucket + create NFT bucket?

Reply:
This is only one action, not two, so we should set nonce+1